src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/Snapshots/RestoreComparer.ProjectPropertyEqualityComparer.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+using NuGet.SolutionRestoreManager;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.NuGet
+{
+    internal static partial class RestoreComparer
+    {
+        private class ProjectPropertyEqualityComparer : EqualityComparer<IVsProjectProperty>
+        {
+            public override bool Equals(IVsProjectProperty x, IVsProjectProperty y)
+            {
+                if (x is null || y is null)
+                    return x == y;
+
+                if (!StringComparers.PropertyNames.Equals(x.Name, y.Name))
+                    return false;
+
+                return StringComparers.PropertyValues.Equals(x.Value, y.Value);
+            }
+
+            public override int GetHashCode(IVsProjectProperty obj)
+            {
+                return obj.Name.GetHashCode();
```

Comment:
Just confirming.. this means GetHashCode can return the same value for two different objects but Equals will return false for those same two objects...

Reply:
Which is a "valid" GetHashCode implementation. GetHashCode determines the bucket you end up in, after which maps/hashtables call Equals on each item in the bucket to find the actual item. If Name clashes were common, then this would be perf issue, the expectation is that they are not.